aioaws/sqs.py

```diff
@@ -0,0 +1,180 @@
+            self._queue_name_or_url = _QueueURL(
+                await self._get_queue_url_from_name_and_region(
+                    self._queue_name_or_url.name,
+                    self._client,
+                    auth=self._auth,
+                )
+            )
+        return self._queue_name_or_url.url
+
+    async def poll(
+        self,
+        *,
+        config: PollConfig = PollConfig(),
+    ) -> AsyncIterator[Iterable[SQSMessage]]:
+        queue_url = await self._get_queue_url()
+        while True:
+            resp = await self._client.get(
+                url=queue_url,
+                params={
+                    "Action": "ReceiveMessage",
+                    "MaxNumberOfMessages": config.max_messages,
+                    "WaitTimeSeconds": config.wait_time,
+                },
+                headers={
+                    "Accept": "application/json",
+                },
+                timeout=Timeout(
+                    5,  # htppx's default timeout
+                    # arbitrary selection of 1.5x wait time
+                    # to avoid http timeouts while long polling
+                    read=config.wait_time * 1.5,
+                ),
+                auth=self._auth,
+            )
+            resp.raise_for_status()
+            yield [
+                SQSMessage.construct(
+                    message_id=message_data["MessageId"],
+                    receipt_handle=message_data["ReceiptHandle"],
+                    md5_of_body=message_data["MD5OfBody"],
+                    body=message_data["Body"],
+                    attributes=message_data["Attributes"],
+                )
+                for message_data in resp.json()["Messages"]
+            ]
+
+    async def change_visibility(self, message: SQSMessage, timeout: int) -> None:
+        queue_url = await self._get_queue_url()
+        if timeout >= MAX_VISIBILITY_TIMEOUT:
+            raise ValueError(
+                f"timeout value range is 0 to {MAX_VISIBILITY_TIMEOUT}, got {timeout}"
+            )
+        await self._client.post(
+            url=queue_url,
+            params={
+                "Action": "ChangeMessageVisibility",
+                "VisibilityTimeout": timeout,
+                "ReceiptHandle": message.receipt_handle,
+            },
+            auth=self._auth,
+            headers={
+                "Accept": "application/json",
+            }
+        )
+    
+    async def delete_message(self, message: SQSMessage) -> None:
+        queue_url = await self._get_queue_url()
+        await self._client.post(
+            url=queue_url,
+            params={
+                "Action": "DeletreMessage",
+                "ReceiptHandle": message.receipt_handle,
+            },
+            auth=self._auth,
+            headers={
+                "Accept": "application/json",
+            },
+        )
+
+
+@asynccontextmanager
+async def create_sqs_client(
```

Comment:
The idea here is to avoid the state of never running `AsyncClient.__aexit__`. This can be done using `SQSClient.__aexit__` but it's not transmittable via the type system and more prone to user error (you can instantiate `SQSClient` without using it as a context manager).